### ISSUE TITLE
[a11y]: Links in data tables are not accessible by keyboard

### ISSUE THREAD

**Issue by jjennings7** [carbon-design-system/carbon#9684]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

7.43.0

### Automated testing tool and ruleset

IBM equal Access Accessibility Checker

### Assistive technology

no

### Description

Using a keyboard and pressing tab, the focus does not stop on links. This prevents the keyboard from being able to click on any links in the data table. 

This issue is present in the common Carbon data table code, and can be viewed in the storybook data table examples that contain links:
https://react.carbondesignsystem.com/?path=/story/components-datatable--with-overflow-menu

### WCAG 2.1 Violation

2.4.3 Focus order 

### CodeSandbox example

https://codesandbox.io/s/yxu99

### Steps to reproduce

This can be observed in the storybook data table examples that contain links:
https://react.carbondesignsystem.com/?path=/story/components-datatable--with-overflow-menu

Regarding the sandbox link - we didn't find any examples of a data table in Sandbox with links, that's why our Code sandbox example does not include links. 

This issue was also noted by @thefirstartist

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by joshblack** [comment]
Hi there @jjennings7! 👋 

I believe this happens in our stories because the links do not have a `href` attribute and are treated by the browser as not interactive. If you place a link with an href in the table, it will receive focus. Here's an example using the CodeSandbox from above: https://codesandbox.io/s/yxu99?file=/index.js

Hope this helps!

**Comment by thefirstartist** [comment]
@joshblack 
That makes sense, Josh. However, this happens not just in the stories where show as links, we also see this in the product and dev claimed that this is Carbon's native behavior. We'll circle back and investigate further. Thanks!

**Comment by joshblack** [comment]
Of course! Happy to help. I'm going to go ahead and close this for now but feel free to comment or make a new issue with the steps to reproduce. Thanks!

**Comment by nsharma1989** [comment]
Hi I am facing the same issue. 

Using version 
```├── carbon-components@10.43.0
├── carbon-components-react@7.43.0
```
Issue 1: Cant access any columns in the dataTable.
Issue 2: Table with expansion ignores anything under it (Using a table).
